### ISSUE TITLE
help(Angular Mat Table): Generating and displaying nested angular material table dynamically

### ISSUE THREAD

**Issue by pravinambekar** [angular/components#23832]
I have came across a situation where i need to implement logic for displaying nested angular mat table. I know and have implemented logic for displaying nested (inner) table with data source child element. But situation is tricky when there is third level child items which again can have nested elements. the example data can be seen below.

![image](https://user-images.githubusercontent.com/41193836/138877967-c1234ecd-9cdd-43fd-8ba7-2348174d7308.png)

So far i have fetched data for parent data source and child data source and displayed using mat table but unable to define logic for generating child mat table at run time based on data receives from backend service. Current implantation is as follow (this is for nested mat table for level 1):

 ```
 <ng-container matColumnDef="expandedDetail">
    <td mat-cell *matCellDef="let element" [attr.colspan]="displayedColumns.length">
      <div class="example-element-detail" *ngIf="element._mdbpartchilds?.length" [@detailExpand]="element == expandedElement ? 'expanded' : 'collapsed'">
        <div class="inner-table mat-elevation-z8" *ngIf="expandedElement">

          <table #innerTables mat-table #innerSort="matSort" [dataSource]="element._mdbpartchilds" matSort>
            <ng-container matColumnDef="{{innerColumn}}" *ngFor="let innerColumn of innerDisplayedColumns">
              <th mat-header-cell *matHeaderCellDef mat-sort-header> {{innerColumn}} </th>
              <td mat-cell *matCellDef="let element"> {{element[innerColumn]}} </td>
            </ng-container>
            <tr mat-header-row *matHeaderRowDef="innerDisplayedColumns"></tr>
            <tr mat-row *matRowDef="let row; columns: innerDisplayedColumns;"></tr>
          </table>
        </div>
      </div>
    </td>
  </ng-container>
  <tr mat-header-row *matHeaderRowDef="displayedColumns"></tr>
  <tr mat-row *matRowDef="let element; columns: displayedColumns;" [class.example-element-row]="element._mdbpartchilds?.length"
   [class.example-expanded-row]="expandedElement === element" (click)="toggleRow(element)">
  </tr>
  <tr mat-row *matRowDef="let row; columns: ['expandedDetail']" class="example-detail-row"></tr>
```
** But assuming if child element also contains some sub child rows how one should implement the logic ? The data for each level can be fetched on row or mat-cell click and data source will get bind to but i see a challenge here to update whole data source when new data arrives for nested table.

To summarize the items can have unpredictable no. of child items which should be displayed if child count is greater than 0. Any help would greatly appreciated.**
  
#### Environment

- Angular 9.1.11:
- CDK/Material: 9.2.4
- Browser(s): Chrome
- Operating System (e.g. Windows, macOS, Ubuntu): Windows

**Comment by crisbeto** [comment]
Please keep GitHub issues for bug reports / feature requests. Better avenues for troubleshooting / questions are stack overflow, gitter, mailing list, etc.

**Comment by pravinambekar** [comment]
> feature request

Hello @crisbeto 
While I completely agree with your comment, I already have attempted to seek solution on SO, unfortunately i didn't found suitable answer so I thought, it could be a feature request or possibility to customize Mat-table to meet my requirement.
Nevertheless , i have to switch to some other custom component to achieve what I was looking for.
Thank you.
